notifications.yaml

```diff
@@ -0,0 +1,220 @@
+        example: 503
+      detail: 
+        type: "string"
+        description: "A human readable explanation specific to this occurrence of the\nproblem.\n"
+        example: "Connection to database timed out"
+      instance: 
+        type: "string"
+        format: "uri"
+        description: "An absolute URI that identifies the specific occurrence of the problem.\nIt may or may not yield further information if dereferenced."
+  MessageContent: 
+    title: "MessageContent"
+    type: "object"
+    properties: 
+      title: 
+        type: "string"
+        minLength: 1
+        maxLength: 255
+        description: "The title of the message. At most 255 characters."
+      body_text: 
+        type: "string"
+        x-faker: "lorem.paragraph"
+      body_html: 
+        type: "string"
+        x-faker: "lorem.paragraph"
+    required: 
+      - "title"
+  MessageDefinition: 
+    title: "MessageDefinition"
+    type: "object"
+    properties: 
+      to: 
+        description: "One fiscal code (CF).\n"
+        type: "string"
+      dry_run: 
+        type: "boolean"
+        description: "This parameter, when set to true, allows developers to test a request without actually sending a message.\n\nThe default value is false."
+      time_to_live: 
+        type: "integer"
+        description: "This parameter specifies how long (in seconds) the message should be kept in storage if the device is offline. The maximum time to live supported is 1 year, and the default value is 1 hour."
+        minimum: 3600
+        maximum: 31536000
+      tags: 
+        type: "array"
+        default: []
+        items: 
+          type: "string"
+      content: 
+        $ref: "#/definitions/MessageContent"
+    required: 
+      - "to"
+      - "content"
+  MessageStatus: 
+    title: "MessageStatus"
+    type: "object"
+    properties: 
+      status: 
+        type: "string"
+        description: "ACCEPTED: the message has been accepted for delivery\n\nDELIVERED: the message was delivered to the channel\n\nFAILED: something wong happened, check the error code and message\n\nTIMED_OUT:  in case a TTL was set\n"
+        enum: 
+          - "ACCEPTED"
+          - "DELIVERED"
+          - "TIMED_OUT"
+          - "FAILED"
+      failed_at: 
+        type: "string"
+        format: "date-time"
+      delivered_at: 
+        type: "string"
+        format: "date-time"
+      accepted_at: 
+        type: "string"
+        format: "date-time"
+      timed_out_at: 
+        type: "string"
+        format: "date-time"
+      problem: 
+        $ref: "#/definitions/ProblemJson"
+  Message: 
+    title: "Message"
+    allOf: 
+      - 
+        type: "object"
+        properties: 
+          id: 
+            type: "string"
+            format: "uuid"
+          created_at: 
+            type: "string"
+            format: "date-time"
+          status: 
+            type: "object"
+            default: {}
+            properties: 
```

Comment:
not sure that the sender needs to know on which channels the message has been received and when, we should not expose the information (at least at this stage) and just tell the sender that the message has been delivered to at least a channel

here you can see the ambiguity between "message" and "notification": 

if we assume that this system provides an "inbox" for the user, the message *always* gets delivered to the user inbox, so as long as sendMessage succeeds, the sender knows the message has been delivered to the user inbox

on the other end, a message being delivered to one's inbox should trigger a notification (hey you got a new message), notifications can contain a preview of the message and gets delivered over multiple channels and multiple times (in case one wants an acknowledge)

I feel like we haven't explored enough the complexity of this kind of interaction (messages vs notifications) - as we want this API to do both (provide an inbox for messages and deliver notifications)